crates/controller/src/hud/minimap/interaction.rs

```diff
@@ -105,15 +119,62 @@ fn click_handler(
     }
 }
 
-fn move_camera_system(
-    mut click_events: EventReader<MinimapClickEvent>,
+fn move_camera_click_system(
+    window_query: Query<&Window, With<PrimaryWindow>>,
+    mut input_events: EventReader<MouseButtonInput>,
+    hud: HudNodes<With<MinimapNode>>,
+    bounds: Res<MapBounds>,
+    mut dragging: ResMut<DraggingCamera>,
     mut camera_events: EventWriter<MoveFocusEvent>,
 ) {
-    for click in click_events.iter() {
-        if click.button() != MouseButton::Left {
+    let Some(cursor) = window_query.single().cursor_position() else {
+        return;
+    };
+
+    for event in input_events.iter() {
+        if event.button != MouseButton::Left {
+            continue;
+        }
+
+        if !event.state.is_pressed() {
+            if dragging.0 {
+                dragging.0 = false;
+            }
             continue;
         }
-        camera_events.send(MoveFocusEvent::new(click.position()));
+
+        if let Some(relative) = hud.relative_position(cursor) {
+            dragging.0 = true;
+            let proportional = Vec2::new(relative.x, 1. - relative.y);
+            let event = MoveFocusEvent::new(bounds.rel_to_abs(proportional));
+
+            camera_events.send(event);
+        }
+    }
+}
+
+fn move_camera_drag_system(
```

Comment:
- I think that dragging implementation is independent on the fix to mouse click / press / release behavior. Therefore, I would separate this changes to a standalone system.
- I would change this system to `dragg_handler()` (similar to `click_handler()`) sending `MinimapDraggEvent` and handle the events in `move_camera_system()`.